discovery_engine_core/core/src/engine.rs

```diff
@@ -337,11 +338,15 @@ where
     pub async fn set_markets(
         &mut self,
         history: &[HistoricDocument],
-        markets: Vec<Market>,
+        new_markets: Vec<Market>,
     ) -> Result<(), Error> {
-        *self.config.markets.write().await = markets;
-
         {
+            let markets = &mut *self.config.markets.write().await;
+            let mut old_markets = replace(markets, new_markets);
```

Comment:
:art: imo it looks a bit cleaner to use the trait instead of de- and re-referencing
```suggestion
            let mut markets = self.config.markets.write().await;
            let mut old_markets = replace(markets.deref_mut(), new_markets);
```